app/controllers/concerns/eventable.rb

```diff
@@ -0,0 +1,38 @@
+module Eventable
+  extend ActiveSupport::Concern
+
+  def supplier_id
+    # NB: not all events will have a supplier_id so this could well be nil
+    @supplier_id ||= current_user.owner&.id
+  end
+
+  def process_event(eventables, event_name, event_params)
```

Comment:
where is `process_event` called from?

Reply:
It's the same as the previous eventable implementations that I'll be removing in a separate PR. It get's included in the relevant controllers (AllocationEventsController, MoveEventsController, JourneyEventsController and invoked as part of each action)